webapp/src/components/App/Studies/LauncherDialog.tsx

```diff
@@ -49,15 +50,15 @@ interface Props {
   onClose: () => void;
 }
 
-function LauncherDialog(props: Props) {
+function LauncherDialog(props: Readonly<Props>) {
   const { studyIds, open, onClose } = props;
   const [t] = useTranslation();
   const { enqueueSnackbar } = useSnackbar();
   const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
   const [options, setOptions] = useState<LaunchOptions>({
     nb_cpu: DEFAULT_NB_CPU,
     auto_unzip: true,
-    time_limit: DEFAULT_TIME_LIMIT,
+    time_limit: undefined,
```

Comment:
Best to initialize with `null`

Reply:
Non, je ne crois pas que ce soit la bonne façon de faire. Je préfère éviter d’intégrer ce type de fichier de configuration du côté frontend, surtout quand un mécanisme existe déjà côté backend. Par ailleurs, il est important de considérer la version Desktop pour laquelle le time_limit n’est pas pertinent : cette valeur est exclusivement utile pour le SLURM et non pour les exécutions locales.

En réalité, c’est la responsabilité du backend de définir les valeurs par défaut, ce qui est le cas ici : le nouveau endpoint récupère la valeur en lisant le fichier `config.yaml`, qui inclut le paramètre `default_time_limit`. Il est impératif que le frontend adopte cette valeur comme limite maximale.

En fait, la valeur fixe `DEFAULT_TIME_LIMIT` dans le frontend s’avère utile uniquement si le endpoint plante (erreur 500). À la limite, on pourrait envisager une valeur nulle. Cependant, si le endpoint est hors service, il y a de fortes chances que l’application soit inutilisable. Cette valeur pourrait également être pertinente lors des tests du frontend en l’absence du backend.

Reply:
Si tu considère que le lancement doit se faire uniquement si le ce endpoint est en service dans ce cas effectivement, on peut s'en passer dans ce cas, tu peux initialiser le state pour `time_limit` a 0.